### PR TITLE
add image publishing job for vsphere csi

### PR DESCRIPTION
**Pull request by divyenpatel**
Add image pushing job for vsphere csi

cc: @xing-yang @XudongLiuHarold

### CODE REVIEW

**Comment by divyenpatel** [comment]
/hold

wait for https://github.com/kubernetes/k8s.io/pull/7210 to merge

**Comment by divyenpatel** [comment]
cc: @upodroid

File: config/jobs/image-pushing/k8s-staging-csi-vsphere.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      spec:
+        serviceAccountName: gcb-builder
+        containers:
+          - image: gcr.io/k8s-staging-test-infra/image-builder:v20240725-9fc4a08b13
+            command:
+              - run.sh
+            args:
+              # this is the project GCB will run in, which is the same as the GCR
+              # images are pushed to.
+              - --project=k8s-infra-staging-csi-vsphere
```
Comment (divyenpatel) [inline config/jobs/image-pushing/k8s-staging-csi-vsphere.yaml:27]: @upodroid is this section looks correct?

Comment (upodroid) [inline config/jobs/image-pushing/k8s-staging-csi-vsphere.yaml:27]:
```suggestion
              - --project=k8s-staging-images
```

File: config/jobs/image-pushing/k8s-staging-csi-vsphere.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+              - --scratch-bucket=gs://k8s-infra-staging-csi-vsphere-gcb
```
Comment (upodroid) [inline config/jobs/image-pushing/k8s-staging-csi-vsphere.yaml:28]:
```suggestion
              - --scratch-bucket=gs://k8s-staging-images-gcb
```

**Review by upodroid — CHANGES_REQUESTED** [review]

File: config/jobs/image-pushing/k8s-staging-csi-vsphere.yaml (hunk 1 above)
Comment (divyenpatel) [reply config/jobs/image-pushing/k8s-staging-csi-vsphere.yaml]: accepted this change in the commit.

**Review by upodroid — APPROVED** [review]
LGTM

**Pull request merged**